synapse/handlers/sso.py

```diff
@@ -702,8 +710,98 @@ async def _register_mapped_user(
         await self._store.record_user_external_id(
             auth_provider_id, remote_user_id, registered_user_id
         )
+
+        # Set avatar, if available
+        if attributes.picture:
+            await self.set_avatar(registered_user_id, attributes.picture)
+
         return registered_user_id
 
+    async def set_avatar(self, user_id: str, picture_https_url: str) -> None:
+        """Set avatar of the user.
+
+        This downloads the image file from the url provided, store that in
+        the media repository and then sets the avatar on user's profile.
```

Comment:
Please also describe
- how this function can fail (HTTP errors; HTTP successes with a bad response; failure to use the media store; anything else??)
-  what this function does if there's a failure (logs a warning and returns early)

Reply:
Done!